servant-subscriber-0.5.3 build failure (due to ambiguity) with latest servant

As seen on the stackage nightly build:

```
src/Servant/Subscriber/Subscribable.hs:98:5: error:
    Ambiguous occurrence <E2><80><98>And<E2><80><99>
    It could refer to either <E2><80><98>Servant.And<E2><80><99>,
                             imported from <E2><80><98>Servant<E2><80><99> at src/Servant/Subscriber/Subscribable.hs:28:1-24
                             (and originally defined in <E2><80><98>Servant.API.TypeLevel
<E2><80><99>)
                          or <E2><80><98>Servant.Subscriber.Subscribable.And<E2><80><99>,
                             defined at src/Servant/Subscriber/Subscribable.hs:97:1

src/Servant/Subscriber/Subscribable.hs:101:5: error:
    Ambiguous occurrence <E2><80><98>Elem<E2><80><99>
    It could refer to either <E2><80><98>Servant.Elem<E2><80><99>,
                             imported from <E2><80><98>Servant<E2><80><99> at src/Servant/Subscriber/Subscribable.hs:28:1-24
                             (and originally defined in <E2><80><98>Servant.API.TypeLevel
<E2><80><99>)
                          or <E2><80><98>Servant.Subscriber.Subscribable.Elem<E2><80><99>,
                             defined at src/Servant/Subscriber/Subscribable.hs:100:1
```

etc.

--- a/src/Servant/Subscriber/Subscribable.hs
+++ b/src/Servant/Subscriber/Subscribable.hs
@@ -90,15 +90,3 @@ instance HasForeign lang ftype sublayout => HasForeign lang ftype (Subscribable 
 instance HasLink sub => HasLink (Subscribable :> sub) where
     type MkLink (Subscribable :> sub) = MkLink sub
     toLink _ = toLink (Proxy :: Proxy sub)
-
--------------- Copied from Servant.Util.Links (they are not exported) ----------
-
--- | If both a or b produce an empty constraint, produce an empty constraint.
-type family And (a :: Constraint) (b :: Constraint) :: Constraint where
-    And () ()     = ()
-
-type family Elem e es :: Constraint where
-    Elem x (x ': xs) = ()
-    Elem y (x ': xs) = Elem y xs
-
---------------------------------------------------------------------------------

